pandas/core/dtypes/dtypes.py

```diff
@@ -671,6 +675,12 @@ def _get_common_dtype(self, dtypes: list[DtypeObj]) -> DtypeObj | None:
 
         return find_common_type(non_cat_dtypes)
 
+    @cache_readonly
```

Comment:
Could you make these properties like the base class?

Reply:
updated to make both cache_readonly; I think the perf improvement in the OP depends on it

Reply:
Updated + greenish